gtests/test_nopython_radixtree.cpp

```diff
@@ -0,0 +1,78 @@
+#include <modmesh/toggle/RadixTree.hpp>
+#include <gtest/gtest.h>
+
+#ifdef Py_PYTHON_H
+#error "Python.h should not be included."
+#endif
+
+template <typename T>
+std::string get_info(modmesh::RadixTreeNode<T> & r)
+{
+    std::ostringstream oss;
+    if (nullptr != r.get_prev())
+    {
+        oss << r.name() << " - " << r.data();
+    }
+    return oss.str();
+}
+
+TEST(RadixTree, construction)
+{
+    namespace mm = modmesh;
```

Comment:
The tests also work as examples for C++ API.  We do not want users to use the library by using the namespace.  Thus, use `mm = modmesh` alias here.